### PR TITLE
Steam: Ignore Franchises that contain a developer or publisher name from being added as series #174

### PR DESCRIPTION
**Pull request by darklinkpower**
Steam: Ignore Franchises that contain a developer or publisher name from being added as series #174

Built and tested

### CODE REVIEW

File: source/Libraries/SteamLibrary/SteamShared/MetadataProvider.cs [hunk 1]
```diff
@@ -445,16 +445,18 @@ internal string ParseDescription(string description)
                     if (ass["type"].Value == "franchise")
                     {
                         var value = ass["name"].Value;
-                        if (publishers.HasItems() && publishers.Any(x => x.Contains(value, StringComparison.OrdinalIgnoreCase)))
+                        if (publishers.HasItems() && (publishers.Any(x => x.Contains(value, StringComparison.OrdinalIgnoreCase) ||
```
Comment (JosefNemec) [inline source/Libraries/SteamLibrary/SteamShared/MetadataProvider.cs]: `publishers.HasItems()` is not necessary if you are going to call `Any` anyways, just add null check `.?Any`.

Comment (darklinkpower) [reply source/Libraries/SteamLibrary/SteamShared/MetadataProvider.cs]: To use a null check, a cast is needed. I personally don't use them but should I change it and use a cast?

image

Comment (JosefNemec) [reply source/Libraries/SteamLibrary/SteamShared/MetadataProvider.cs]: You need to add check specifically for `true`: `publishers?.Any(x => x.Contains(value, StringComparison.OrdinalIgnoreCase) == true`

File: source/Libraries/SteamLibrary/SteamShared/MetadataProvider.cs [hunk 2]
```diff
@@ -451,6 +468,16 @@ internal string ParseDescription(string description)
             return metadata;
         }
 
+        private static IEnumerable<string> GetCompaniesFromAppDetails(List<string> companies)
```
Comment (JosefNemec) [inline source/Libraries/SteamLibrary/SteamShared/MetadataProvider.cs:471]: This looks unnecessary to me, you can replace this with: `publishers = metadata.StoreDetails.publishers?.Where(a => !a.IsNullOrWhiteSpace())`

Also, are they really including empty publishers and developers? Is that check even necessary?

Comment (darklinkpower) [reply source/Libraries/SteamLibrary/SteamShared/MetadataProvider.cs:471]: I'm not sure to be honest. The extension currently checks that so I kept it for safety

https://github.com/JosefNemec/PlayniteExtensions/blob/044ba2ab70891833b1dfb7c25998b71f7f57e390/source/Libraries/SteamLibrary/SteamShared/MetadataProvider.cs#L304-L309

Comment (darklinkpower) [reply source/Libraries/SteamLibrary/SteamShared/MetadataProvider.cs]: Changed to use linq in https://github.com/JosefNemec/PlayniteExtensions/pull/175/commits/f360f8e03e16abf85c54a30ecaedcfb651711c3d

**Pull request merged**